lib/plugins/aws/package/compile/events/eventBridge/index.js

```diff
@@ -190,30 +148,52 @@ class AwsCompileEventBridgeEvents {
           }
         });
       }
-
-      if (funcUsesEventBridge) {
-        iamRoleStatements.push({
-          Effect: 'Allow',
-          Resource: {
-            'Fn::Join': [
-              ':',
-              [
-                'arn',
-                { Ref: 'AWS::Partition' },
-                'lambda',
-                { Ref: 'AWS::Region' },
-                { Ref: 'AWS::AccountId' },
-                'function',
-                FunctionName,
-              ],
-            ],
-          },
-          Action: ['lambda:AddPermission', 'lambda:RemovePermission'],
-        });
-      }
     });
 
-    iamRoleStatements.unshift(...eventBusIamRoleStatements.values());
+    if (anyFuncUsesEventBridge) {
+      const ruleResources = {
+        'Fn::Join': [
+          ':',
+          [
+            'arn',
+            { Ref: 'AWS::Partition' },
+            'events',
+            { Ref: 'AWS::Region' },
+            { Ref: 'AWS::AccountId' },
+            'rule/*',
+          ],
+        ],
+      };
+      iamRoleStatements.push({
+        Effect: 'Allow',
+        Resource: ruleResources,
+        Action: [
+          'events:PutRule',
+          'events:RemoveTargets',
+          'events:PutTargets',
+          'events:DeleteRule',
+        ],
+      });
+      const functionResources = {
```

Comment:
Shouldn't this be applied only if at least one event bridge event referenced event bus?

At least this will match original implementation

Reply:
If I'm reading the code correctly, it was adding `lambda:AddPermission` and `lambda:RemovePermission` for each function that uses EventBridge event (`funcUsesEventBridge` flag). In the new version, we add this rule if *at least one function users EventBridge event*, which is the same logic.

If you mean `events:*` permissions, it also seems like we have added them if at least one function uses any event bridge events.

Reply:
Thanks for explanations, I see it's ok